src/Aacotroneo/Saml2/Http/Controllers/Saml2Controller.php

```diff
@@ -100,4 +103,146 @@ public function login()
         $this->saml2Auth->login(config('saml2_settings.loginRoute'));
     }
 
+    /**
+     * Gets the HL7 message from the user request
+     *
+     * @param   Saml2User     $user
+     * @return  String       
+     */
+    private function getHL7MessageFromRequest( $user )
+    {
+        $attributes = $user->getAttributes();
+
+        if( !array_key_exists('RequestMessage', $attributes) || sizeof($attributes['RequestMessage']) === 0 )
+        {
+            return null;
+        }
+
+        return $attributes['RequestMessage'][0];
+    }
+
+    /**
+     * Get the user model using the email on request
+     *
+     * @param   Saml2User   $user
+     * @return  User
+     */
+    private function getUserFromRequest( $samlUser )
+    {
+        $email = $samlUser->getAttributes()['Email'][0];
+
+        // Find a node with the attribute Name set as Email, after find the text node that contains the email
+        $user = User::whereUsername( $email )->first();
+        if ( is_null($user) )
+        {
+            throw new Saml2UserNotPresentException( "Saml request with username {$username} does not have an user on AristaMD" );
+        }
+        return $user;
+    }
+
+    /**
+     * Get the query params for the request object
+     *
+     * @param   Request   $request
+     * @return  String
+     */
+    public function getUrlParamString( $request )
+    {
+        if( empty($request) || empty($request->id) )
+        {
+            return "";
+        }
+
+        $recordType = null;
+
+        switch ( get_class($request) )
+        {
+            case 'App\Models\EConsult':
+                $recordType = 'econsult';
+                break;
+            case 'App\Models\Referral':
+                $recordType = 'referral';
+                break;
+            default:
+                $recordType = null;
+                break;
+        }
+
+        return "&record_type=$recordType&record_id=$request->id";
+    }
+
+    /**
+     * Creates a referral or an eConsult from a HL7 message
+     *
+     * @param   String  $message    HL7 message as string
+     * @param   User    $user       Authenticated user
+     * @return  Integer             Request id
+     */
+    private function createRequest( $message, $user )
+    {
+        if ( empty($message) )
+        {
+            /*$this->responseCode( 500 );
+            $this->message = "Internal Error. Please provide the HL7 v.2.3 REF message.";
```

Comment:
Do we need this comment here? in case that is not required, Could we remove it? If the condition is not needed please remove it too.